src/makeCalculator.js

```diff
@@ -4,7 +4,41 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+
+    add: function (a, b) {
+      return a + b;
+    },
+
+    subtract: function (a, b) {
+      return a - b;
+    },
+
+    multiply: function (a, b) {
+      return a * b;
+    },
+
+    divide: function (a, b) {
+      if (b !== 0) {
+        return a / b;
+      } else {
+        throw new Error('Division by zero is not allowed.');
+      }
```

Comment:
[CODE KNOWLEDGE] - The `divide` method should be defined using the shorthand syntax as well. Convert `divide: function (a, b) { ... }` to `divide(a, b) { ... }`.